### PR TITLE
test(appium): adding files tests and helpers

### PR DESCRIPTION
**Pull request by luisecm**
### What this PR does 📖
- Adding the first files tests for Uplink app
- Adding helper function to retrieve custom predicate strings locators
- Adding screenobject for files
- Updated WDIO CI config file to include xcode build logs in order to track proxy errors presented randomly
- Updated WDIO shared config file to use the proper test results subfolders depending on driver capabilities (macos, windows, ubuntu)
- Updated appium workflow file to include the proper folder on failed tests screenshots artifact

### Which issue(s) this PR fixes 🔨
- Resolve #

### Special notes for reviewers 🗒️
- This is a work in progress, but sending now as PR since I want to give it a try on CI

### CODE REVIEW

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
🎉

File: appium-tests/tests/specs/files-screen.spec.ts [hunk 1]
```diff
@@ -19,19 +19,21 @@ describe("Files Screen Tests on Uplink Desktop", async () => {
...
     await expect(await FilesScreen.availableSpaceIndicatorText).toBeDisplayed()
     await expect(
       await FilesScreen.availableSpaceIndicatorText,
-    ).toHaveTextContaining("Free")
+    ).toHaveTextContaining(/[. 0-9] Free/i)
     await expect(await FilesScreen.usedSpaceIndicatorText).toBeDisplayed()
     await expect(await FilesScreen.usedSpaceIndicatorText).toHaveTextContaining(
-      "GB / ",
-    )*/
+      [/[. 0-9]+(KB|MB|GB) /, "/ ", /. 0-9]+(KB|MB|GB)/i],
```
Comment (stavares843) [inline appium-tests/tests/specs/files-screen.spec.ts:35]: neat 💥

Comment (luisecm) [reply appium-tests/tests/specs/files-screen.spec.ts:35]: Thanks!!! @stavares843

**Comment by luisecm** [comment]
Closing this PR since it is no longer needed

**Pull request closed without merging**